compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -608,6 +662,41 @@ impl<'a, 'tcx> ExprUseVisitor<'a, 'tcx> {
             ty::Closure(..) | ty::Generator(..)
         );
 
+        if let Some(fake_reads) = self.mc.typeck_results.closure_fake_reads.get(&closure_def_id) {
+            for (fake_read, cause) in fake_reads.iter() {
+                match fake_read.base {
+                    PlaceBase::Upvar(upvar_id) => {
+                        if upvars.map_or(body_owner_is_closure, |upvars| {
+                            !upvars.contains_key(&upvar_id.var_path.hir_id)
+                        }) {
+                            // The nested closure might be fake reading the current (enclosing) closure's local variables.
+                            // We only want to fake read the fake read present in the nested closure that are not part of
```

Comment:
Maybe "The only places we want to fake read before creating the parent closure are the ones that are not local to it/ defined by it."